src/senaite/core/patches/cmfcore/portal_catalog_processor.py

```diff
@@ -0,0 +1,58 @@
+# -*- coding: utf-8 -*-
+#
+# This file is part of SENAITE.CORE.
+#
+# SENAITE.CORE is free software: you can redistribute it and/or modify it under
+# the terms of the GNU General Public License as published by the Free Software
+# Foundation, version 2.
+#
+# This program is distributed in the hope that it will be useful, but WITHOUT
+# ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS
+# FOR A PARTICULAR PURPOSE. See the GNU General Public License for more
+# details.
+#
+# You should have received a copy of the GNU General Public License along with
+# this program; if not, write to the Free Software Foundation, Inc., 51
+# Franklin Street, Fifth Floor, Boston, MA 02110-1301 USA.
+#
+# Copyright 2018-2024 by it's authors.
+# Some rights reserved, see README and LICENSE.
+
+from bika.lims import api
+
+PORTAL_CATALOG = "portal_catalog"
+
+
+def index_in_portal_catalog(obj):
```

Comment:
Shouldn't we also check for `api.is_temporary(obj)`, like what you did in `catalog_multiplex`?

Reply:
Everything that ends up in this queue comes from `Products.CMFCore.CatalogTool.reindexObject` which does already `filterTemporaryItems`, which uses our patched `isTemporary` for both AT and DX types:

https://github.com/senaite/senaite.core/blob/c725d9d0c0652fd0e0b013f16154ede0be6d5b99/src/senaite/core/patches/archetypes/__init__.py#L26-L29

https://github.com/senaite/senaite.core/blob/c725d9d0c0652fd0e0b013f16154ede0be6d5b99/src/senaite/core/patches/dexterity/dexterity_content.py#L29-L30

So another `api.is_temporary` is at this stage not required